### PR TITLE
Create plugin: Fix update command

### PR DESCRIPTION
**Pull request by leventebalogh**
> [!NOTE]  
> This PR does not focus on refactoring, it is only focusing on putting the fires out. Refactoring (e.g. for template data) is planned to happen in an upcoming PR.

### The Problem
There was a bug introduced in `v4.0.0` which will stop externalising the `react-router-dom` dependency when updating existing plugins, and this breaks the UI, especially if they don't have any of the `.cprc.json` files. (**The bug only affects app plugins.**) 

Digging into how to fix this issue revealed other bugs as well that we didn't know about: the `templateData` used when updating plugins didn't have the same fields as the one we are using in plugin generation, and this resulted in our templates working incorrectly.

### What changed?
- Use different configs during _scaffolding_ and during _updating_ (This was necessary because there are feature flags that we would like to enable by default for new scaffolds, but we don't want to enable them for existing plugins that doesn't have one (or both) of the `.cprc.json` files. This can be refactored later.
- Start adding debug messages that can be displayed e.g. `DEBUG=*` - at this point it's for template data

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@4.0.1-canary.746.3125322.0
  # or 
  yarn add @grafana/create-plugin@4.0.1-canary.746.3125322.0
  ```
</details>

### CODE REVIEW

**Review by jackw — COMMENTED** [review]
> Use different configs during scaffolding and during updating (This was necessary because there are feature flags that we would like to enable by default for new scaffolds, but we don't want to enable them for existing plugins that doesn't have one (or both) of the .cprc.json files. 

I'm not sure I understand this logic. I thought the changes to the update command would make it so, once run, it would align the configs and dependencies as thought it was a newly scaffolded plugin? However this sentence means that if I created a plugin with create-plugin@2.5.x I will never get react-routerV6 upgrade as the cprc file will never exist? 🤔 

If they don't exist on update should we be writing them both to disk?

File: packages/create-plugin/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  hasGithubLevitateWorkflow: boolean;
+};
+
+export type TemplateData = {
+  pluginId: string;
+  packageManagerName: string;
+  packageManagerInstallCmd: string;
+  packageManagerVersion: string;
+  isAppType: boolean;
+  isNPM: boolean;
```
Comment (academo) [inline packages/create-plugin/src/types.ts:19]: do we really need this flag since we already have `packageManagerName` ?

Comment (jackw) [reply packages/create-plugin/src/types.ts:19]: Due to the way NPM passes flags to run other NPM scripts it's needed for:

```
"lint:fix": "{{ packageManagerName }} run lint{{#if isNPM}} --{{/if}} --fix",
```

The alternative is to remove it in favour of duplicating the entire `lint` script and addind the extra param.

File: packages/create-plugin/src/utils/utils.templates.ts [hunk 2]
```diff
@@ -78,26 +82,34 @@ export function renderTemplateFromFile(templateFile: string, data?: any) {
   return renderHandlebarsTemplate(fs.readFileSync(templateFile).toString(), data);
 }
 
-export function getTemplateData() {
+export function getTemplateData(): TemplateData {
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.templates.ts:85]: Exactly, let's do that 👍

File: packages/create-plugin/src/utils/utils.config.ts [hunk 3]
```diff
@@ -19,24 +21,49 @@ type UserConfig = {
 };
 
 export function getConfig(): CreatePluginConfig {
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.config.ts:23]: `cosmiconfig` sounds like a good thing to explore, I didn't hear about it yet! But yeah, I was thinking the same, adding tests for these is crucial, as there are so many edge-cases that we will forget to test. (And as soon as we have tests it will be much easier to refactor as well, e.g. to use cosmiconfig.)

File: packages/create-plugin/src/utils/utils.config.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import fs from 'node:fs';
 import path from 'node:path';
 import { getVersion } from './utils.version.js';
+import { commandName } from './utils.cli.js';
+import { features } from 'node:process';
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.config.ts:5]: Well spotted mate, thanks! Will remove it. 👍

**Comment by leventebalogh** [comment]
> I'm not sure I understand this logic. I thought the changes to the update command would make it so, once run, it would align the configs and dependencies as thought it was a newly scaffolded plugin? However this sentence means that if I created a plugin with [create-plugin@2.5.x](mailto:create-plugin@2.5.x) I will never get react-routerV6 upgrade as the cprc file will never exist? 🤔 If they don't exist on update should we be writing them both to disk?

Yeah, I know it's getting confusing, in fact these scanarios are not trivial.

Regarding migrating to react-router v6 we thought the following:
- **scaffolding a new plugin:** we already give you react-router v6, as it doesn't cause any extra work for you and it's gonna get you in a better place. This means that we have to enable the feature flag by default.
- **updating:** we still wouldn't like to force plugin devs to do the migration in case they would like to update their plugin, so we wouldn't enable the feature flag by default. We can generate the `root/.cprc.json` for them, but with a different value for feature flags compared to new scaffolds.

Based on the above I think the only difference between _scaffolding_ and _updating_ is going to be in feature flags. If you generate a new plugin, we can bootstrap you an opinionated set of features, while if you are updating we would like to focus on break as few things in your workflow as possible. 

Does this make more sense? :)

File: packages/create-plugin/src/utils/utils.config.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.config.ts]: Fixed in https://github.com/grafana/plugin-tools/pull/746/commits/3125322a1b01337367eaa43725fbd81763f2ef87.

File: packages/create-plugin/src/utils/utils.config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.config.ts:22]: Created an issue for it on our board: https://github.com/grafana/grafana-plugins-platform-team/issues/81

File: packages/create-plugin/src/utils/utils.templates.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (leventebalogh) [reply packages/create-plugin/src/utils/utils.templates.ts:85]: Created an issue for it: https://github.com/grafana/grafana-plugins-platform-team/issues/82

**Comment by jackw** [comment]
>Based on the above I think the only difference between scaffolding and updating is going to be in feature flags. If you generate a new plugin, we can bootstrap you an opinionated set of features, while if you are updating we would like to focus on break as few things in your workflow as possible.

Does this make more sense? :)

Yep, this summary makes perfect sense to me. Thanks for taking the time to write it down so I can reason more easily about the code changes. 🥇

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: packages/create-plugin/src/utils/utils.config.ts [hunk 5]
```diff
@@ -52,9 +78,19 @@ function getUserConfig(): UserConfig | undefined {
         ...userConfig!.features,
       }),
     };
+    // Most likely this happens because of no ".cprc.json" (user configuration) file.
+    // (This can both happen for new scaffolds and for existing plugins that have not been updated yet.)
   } catch (error) {
+    // Scaffolding a new plugin
+    if (commandName === 'generate') {
```
Comment (academo) [inline packages/create-plugin/src/utils/utils.config.ts:85]: NIT: move `generate` to a constant to avoid a magic string

**Review by academo — APPROVED** [review]

**Comment by briangann** [comment]
i did try this release (v4.0.1) and I'm still getting similar scene-app errors:

```SHELL
WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/components/SceneApp/SceneApp.js 15:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 65:0-74 65:0-74 65:0-74
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/components/SceneApp/SceneAppPage.js 127:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 66:0-69 66:0-69
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/core/PanelBuilders/FieldConfigBuilders.js 70:40-65
export 'defaultScatterFieldConfig' (imported as 'defaultScatterFieldConfig') was not found in '@grafana/schema/dist/esm/raw/composable/xychart/panelcfg/x/XYChartPanelCfg_types.gen' (possible exports: ScatterShow, SeriesMapping, defaultFieldConfig, defaultOptions, defaultXYDimensionConfig, pluginVersion)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 73:0-82 73:0-82
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/core/PanelBuilders/index.js 110:12-37
export 'defaultScatterFieldConfig' (imported as 'defaultScatterFieldConfig') was not found in '@grafana/schema/dist/esm/raw/composable/xychart/panelcfg/x/XYChartPanelCfg_types.gen' (possible exports: ScatterShow, SeriesMapping, defaultFieldConfig, defaultOptions, defaultXYDimensionConfig, pluginVersion)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 69:0-62 69:0-62
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ./components/Routes/Routes.tsx 8:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ./components/Routes/index.tsx 1:0-25 1:0-25
 @ ./components/App/App.tsx 3:0-35 19:45-51
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ./components/Routes/Routes.tsx 14:42-50
export 'Redirect' (imported as 'Redirect') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ./components/Routes/index.tsx 1:0-25 1:0-25
 @ ./components/App/App.tsx 3:0-35 19:45-51
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

webpack 5.89.0 compiled with 6 warnings in 398 ms
Type-checking in progress...
assets by status 5.05 MiB [cached] 6 assets
cached modules 2.42 MiB (javascript) 1.25 KiB (runtime) [cached] 1118 modules

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/components/SceneApp/SceneApp.js 15:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 65:0-74 65:0-74 65:0-74
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/components/SceneApp/SceneAppPage.js 127:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 66:0-69 66:0-69
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/core/PanelBuilders/FieldConfigBuilders.js 70:40-65
export 'defaultScatterFieldConfig' (imported as 'defaultScatterFieldConfig') was not found in '@grafana/schema/dist/esm/raw/composable/xychart/panelcfg/x/XYChartPanelCfg_types.gen' (possible exports: ScatterShow, SeriesMapping, defaultFieldConfig, defaultOptions, defaultXYDimensionConfig, pluginVersion)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 73:0-82 73:0-82
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/core/PanelBuilders/index.js 110:12-37
export 'defaultScatterFieldConfig' (imported as 'defaultScatterFieldConfig') was not found in '@grafana/schema/dist/esm/raw/composable/xychart/panelcfg/x/XYChartPanelCfg_types.gen' (possible exports: ScatterShow, SeriesMapping, defaultFieldConfig, defaultOptions, defaultXYDimensionConfig, pluginVersion)
 @ ../node_modules/.pnpm/@grafana+scenes@3.3.0_@grafana+data@10.3.3_@grafana+runtime@10.3.3_@grafana+schema@10.3.3_@gr_5yrxfzfflu6ac5s74djix6z3zu/node_modules/@grafana/scenes/dist/esm/index.js 69:0-62 69:0-62
 @ ./components/App/App.tsx 5:0-45 9:12-48
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ./components/Routes/Routes.tsx 8:45-51
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ./components/Routes/index.tsx 1:0-25 1:0-25
 @ ./components/App/App.tsx 3:0-35 19:45-51
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

WARNING in ./components/Routes/Routes.tsx 14:42-50
export 'Redirect' (imported as 'Redirect') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_FetchersContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_ViewTransitionContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_usePrompt, unstable_useViewTransitionState, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useBlocker, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
 @ ./components/Routes/index.tsx 1:0-25 1:0-25
 @ ./components/App/App.tsx 3:0-35 19:45-51
 @ ./components/App/index.tsx 1:0-22 1:0-22
 @ ./module.ts 2:0-39 4:50-53

webpack 5.89.0 compiled with 6 warnings in 631 ms
Type-checking in progress...
ERROR in ./src/components/Routes/Routes.tsx:2:10
TS2724: '"react-router-dom"' has no exported member named 'Redirect'. Did you mean 'redirect'?
    1 | import * as React from 'react';
  > 2 | import { Redirect, Route, Switch } from 'react-router-dom';
      |          ^^^^^^^^
    3 | import { SentryHomePage } from '../../pages/SentryHome';
    4 | import { SentryConfigPage } from '../../pages/SentryConfig';
    5 | import { prefixRoute } from '../../utils/utils.routing';

ERROR in ./src/components/Routes/Routes.tsx:2:27
TS2305: Module '"react-router-dom"' has no exported member 'Switch'.
    1 | import * as React from 'react';
  > 2 | import { Redirect, Route, Switch } from 'react-router-dom';
      |                           ^^^^^^
    3 | import { SentryHomePage } from '../../pages/SentryHome';
    4 | import { SentryConfigPage } from '../../pages/SentryConfig';
    5 | import { prefixRoute } from '../../utils/utils.routing';

ERROR in ./src/components/Routes/Routes.tsx:11:57
TS2322: Type '{ path: string; component: () => Element; }' is not assignable to type 'IntrinsicAttributes & RouteProps'.
  Property 'component' does not exist on type 'IntrinsicAttributes & RouteProps'. Did you mean 'Component'?
     9 |   return (
    10 |     <Switch>
  > 11 |       <Route path={prefixRoute(`${ROUTES.SentryHome}`)} component={SentryHomePage} />
       |                                                         ^^^^^^^^^
    12 |       <Route path={prefixRoute(`${ROUTES.SentryConfig}`)} component={SentryConfigPage} />
    13 |       <Redirect to={prefixRoute(ROUTES.SentryHome)} />
    14 |     </Switch>

ERROR in ./src/components/Routes/Routes.tsx:12:59
TS2322: Type '{ path: string; component: () => Element; }' is not assignable to type 'IntrinsicAttributes & RouteProps'.
  Property 'component' does not exist on type 'IntrinsicAttributes & RouteProps'. Did you mean 'Component'?
    10 |     <Switch>
    11 |       <Route path={prefixRoute(`${ROUTES.SentryHome}`)} component={SentryHomePage} />
  > 12 |       <Route path={prefixRoute(`${ROUTES.SentryConfig}`)} component={SentryConfigPage} />
       |                                                           ^^^^^^^^^
    13 |       <Redirect to={prefixRoute(ROUTES.SentryHome)} />
    14 |     </Switch>
    15 |   );

Found 4 errors in 2079 ms.
```

adding the external to webpack.config.js for react-router-dom fixes the problem.

**Comment by briangann** [comment]
note - running create plugin with v4.0.1 generates a scenes-app with 
`"react-router-dom": "v5.3.3",`

 but the update sets it to ` "react-router-dom": "^6.22.0",`

to duplicate, generate a scenes-app, then do an update right after, the same type of errors will occur.,

**Comment by leventebalogh** [comment]
(@briangann looks like the [last fix](https://github.com/grafana/plugin-tools/pull/754) is fixing the issue, thanks again for the help.)
